### PR TITLE
Autodeploy config script fixes

### PR DESCRIPTION
**Pull request by AronNovak**
Travis CLI needs GitHub personal access token to authenticate for a while.

<a href="https://gitpod.io/#https://github.com/Gizra/drupal-starter/pull/271"></a>

### CODE REVIEW

**Comment by AronNovak** [comment]
Less head-scratching for the next project init.

File: RoboFile.php [hunk 1]
```diff
@@ -632,14 +632,17 @@ public function phpcs(): ?ResultData {
    *
    * @param string $token
    *   Terminus machine token: https://pantheon.io/docs/machine-tokens.
+   * @param string $github_token
+   *   Personal GitHub token (Travis auth):
+   *   https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token
    * @param string $github_deploy_branch
    *   The branch that should be pushed automatically to Pantheon.
```
Comment (amitaibu) [inline RoboFile.php:639]: Let's mention the default values

File: RoboFile.php [hunk 2]
```diff
@@ -632,14 +632,17 @@ public function phpcs(): ?ResultData {
...
    * @param string $pantheon_deploy_branch
    *   The branch at the artifact repo that should be the target of the deployment.
    *
    * @throws \Exception
    */
-  public function deployConfigAutodeploy(string $token, string $github_deploy_branch = 'master', string $pantheon_deploy_branch = 'master'): void {
+  public function deployConfigAutodeploy(string $token, string $github_token, string $github_deploy_branch = 'main', string $pantheon_deploy_branch = 'qa'): void {
```
Comment (amitaibu) [inline RoboFile.php:645]: Does it need some info on the README? And/ or link to where we call this from Travis

Comment (AronNovak) [reply RoboFile.php:645]: yup, i extended the README for the autodeploy config part.

File: RoboFile.php (hunk 1 above)
Comment (AronNovak) [reply RoboFile.php]: Done

**Comment by amitaibu** [comment]
@AronNovak if you can have a loot at Travis please

image

**Comment by AronNovak** [comment]
It is due to:
```
dealerdirect/phpcodesniffer-composer-installer contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins

You can run "composer config --no-plugins allow-plugins.dealerdirect/phpcodesniffer-composer-installer [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
```

I am on it to address it.

**Comment by AronNovak** [comment]
@amitaibu Shall we merge this then now?

**Comment by amitaibu** [comment]
Thanks

**Pull request merged**